src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -6144,27 +6134,21 @@ public void SetProperty(string path, PSObject propertyToSet)
 
             PSObject results = new PSObject();
             PSObject fileSystemInfoShell = null;
-            bool isContainer = false;
 
-            // Create a PSObject with either a DirectoryInfo or FileInfo object
-            // at its core.
+            var fsinfo = GetFileSystemInfo(path, out bool isDirectory);
 
-            if (Utils.ItemExists(path, out bool isDirectory))
+            // Create a PSObject with either a DirectoryInfo or FileInfo object at its core.
+            if (fsinfo != null)
             {
                 if (isDirectory)
                 {
-                    isContainer = true;
-                    fileSystemInfoShell = PSObject.AsPSObject(new DirectoryInfo(path));
+                    fileSystemInfoShell = PSObject.AsPSObject((DirectoryInfo)fsinfo);
                 }
                 else
                 {
-                    // Maybe the path is a file name so try a FileInfo instead
-                    fileSystemInfoShell = PSObject.AsPSObject(new FileInfo(path));
+                    fileSystemInfoShell = PSObject.AsPSObject((FileInfo)fsinfo);
                 }
-            }
 
-            if (fileSystemInfoShell != null)
-            {
```

Comment:
`fileSystemInfoShell` still could be null, right?

Reply:
Why? We call PSObject.AsPSObject(fsinfo) where fsinfo != null.

Reply:
Ah, you moved the code within `if (fsinfo != null)`. I missed that. #Closed